src/Core/ProjectFilesWatcher.cs

```diff
@@ -0,0 +1,203 @@
+                    value.ProjectChanged += OnProjectChanged;
+                }
+                this.Project = Decompiler?.Project;
+            }
+        }
+
+        public Project? Project
+        {
+            get => project;
+            set
+            {
+                if (project != null)
+                {
+                    RemoveScripts(project.ScriptFiles.ToList());
+                    project.ScriptFiles.CollectionChanged -= OnScriptsChanged;
+                }
+                project = value;
+                if (value != null)
+                {
+                    AddScripts(value.ScriptFiles.ToList());
+                    value.ScriptFiles.CollectionChanged += OnScriptsChanged;
+                }
+            }
+        }
+
+        private void AddScripts(IList newScripts)
+        {
+            foreach(ScriptFile script in newScripts)
+            {
+                var directoryName = Path.GetDirectoryName(script.Filename);
+                var fileName = Path.GetFileName(script.Filename);
+                var watcher = new FileSystemWatcher(directoryName, fileName);
+                if (scriptWatchers.TryAdd(script.Filename, watcher))
+                {
+                    watcher.NotifyFilter = NotifyFilters.LastWrite;
+                    watcher.Changed += OnScriptFileChanged;
+                    watcher.EnableRaisingEvents = true;
+                }
+                else
+                {
+                    watcher.Dispose();
+                }
+            }
+        }
+
+        private void RemoveScripts(IList oldScripts)
+        {
+            foreach (ScriptFile script in oldScripts)
+            {
+                if (scriptWatchers.TryRemove(script.Filename, out var watcher))
+                {
+                    watcher.Changed -= OnScriptFileChanged;
+                    watcher.EnableRaisingEvents = false;
+                    watcher.Dispose();
+                }
+            }
+        }
+
+        private void ReloadScript(ScriptFile scriptFile)
+        {
+            var fileName = scriptFile.Filename;
+            try
+            {
+                var bytes = loader.LoadImageBytes(fileName, 0);
+                var stream = new MemoryStream(bytes);
+                using var rdr = new StreamReader(stream);
+                scriptFile.Evaluate(rdr.ReadToEnd());
+            }
+            catch(Exception ex)
+            {
+                eventListener.Error(ex, $"Failed to load script {fileName}.");
+            }
+        }
+
+        private void OnScriptFileChanged(object sender, FileSystemEventArgs e)
+        {
+            var fullPath = e.FullPath;
+            if (!changedFiles.TryAdd(fullPath, true))
+                return;
+            // Multiple Changed event can be raised in some cases. And file
+            // access error can be occurred when processing first event. Use
+            // timer to delay processing event.
+            var timer = new Timer(200)
```

Comment:
Let's move the 200 into a constant, like `const int MsecEventDelay = 200;`

Reply:
Done in f0beb64